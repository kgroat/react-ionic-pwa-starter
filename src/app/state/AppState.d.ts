
import { RouterState } from 'react-router-redux'

import { AuthState } from './AuthState'
import { Entities } from './Entities'
import { UiState } from './UiState'

export interface AppState {
  auth: AuthState
  router: RouterState
  entities: Entities
  ui: UiState
}
