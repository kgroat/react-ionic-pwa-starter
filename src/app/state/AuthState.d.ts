
export interface AuthState {
  token: string | null
}
