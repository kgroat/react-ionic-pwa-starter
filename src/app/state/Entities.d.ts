
import { User } from 'models/user'
import { FlatMap } from './FlatMap'

export interface Entities {
  users: FlatMap<User>
}
