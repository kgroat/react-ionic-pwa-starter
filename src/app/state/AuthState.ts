
export interface AuthState {
  token: string | null
}

export const DEFAULT_STATE: AuthState = {
  token: null,
}
