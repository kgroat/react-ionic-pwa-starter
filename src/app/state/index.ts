
export { AppState, DEFAULT_STATE } from './AppState'
