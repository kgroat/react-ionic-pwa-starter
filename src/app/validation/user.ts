
import { makeAsyncValidator } from 'shared/validation'
import { User } from 'models/user'

import { GET } from 'redux-store/api'

export { verifyPassword, verifyUsername, verifyEmail } from 'shared/validation/user'

export const verifyUsernameDoesntExists = makeAsyncValidator<string>({
  required: [async (username) => {
    try {
      await GET<User>(`/user/${username}`)
      return `Username "${username}" taken`
    } catch (err) {
      // Do nothing, the username does not already exist
    }
  }],
})

export const verifyEmailNotTaken = makeAsyncValidator<string>({
  required: [async (email) => {
    // Make request to check if email is taken
    if (!email) {
      return 'A user with that email already exists'
    }
  }]
})
