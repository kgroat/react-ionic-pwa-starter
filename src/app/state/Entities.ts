
import { User } from 'models/user'
import { FlatMap } from './FlatMap'

export interface Entities {
  users: FlatMap<User>
}

export const DEFAULT_STATE: Entities = {
  users: {},
}
