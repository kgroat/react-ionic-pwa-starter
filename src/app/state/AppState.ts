
import { RouterState } from 'react-router-redux'

import { AuthState, DEFAULT_STATE as auth } from './AuthState'
import { Entities, DEFAULT_STATE as entities } from './Entities'
import { UiState, DEFAULT_STATE as ui } from './UiState'

export interface AppState {
  router: RouterState
  auth: AuthState
  entities: Entities
  ui: UiState
}

export const DEFAULT_STATE: AppState = {
  router: undefined as any,
  auth,
  entities,
  ui,
}
