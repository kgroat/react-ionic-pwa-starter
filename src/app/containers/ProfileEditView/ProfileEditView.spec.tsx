
import * as React from 'react'
import { create } from 'react-test-renderer'
import { MemoryRouter, Route } from 'react-router'

import * as reduxMockStore from 'redux-mock-store'
const createMockStore: typeof reduxMockStore.default = reduxMockStore as any
import thunk from 'redux-thunk'

import { User } from 'models/user'
import { AppState, DEFAULT_STATE } from 'state'

import ConnectedProfileEditView, { ProfileEditView } from './ProfileEditView'

const mockStore = createMockStore<AppState>([thunk])
jest.mock('containers/ContentView')

const user: User = {
  username: 'testuser',
  firstName: 'Test',
  lastName: 'McUser',
  email: 'test@user.com',
  verified: true,
}

describe(`<${ProfileEditView.name} />`, () => {
  describe('snapshots', () => {
    it('should render loading when the user is undefined', () => {
      const store = mockStore({
        ...DEFAULT_STATE,
        entities: { users: { me: undefined } },
      })

      const tree = create((
        <MemoryRouter>
          <Route render={props => <ConnectedProfileEditView store={store} {...props} />} />
        </MemoryRouter>
      ))

      expect(tree).toMatchSnapshot()
    })

    it('should render the details of the user', () => {
      const store = mockStore({
        ...DEFAULT_STATE,
        entities: { users: { me: user } },
      })

      const tree = create((
        <MemoryRouter>
          <Route render={props => <ConnectedProfileEditView store={store} {...props} />} />
        </MemoryRouter>
      ))

      expect(tree).toMatchSnapshot()
    })
  })
})
