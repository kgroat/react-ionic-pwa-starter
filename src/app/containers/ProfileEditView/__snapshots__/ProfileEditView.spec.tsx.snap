// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileEditView /> snapshots should render loading when the user is undefined 1`] = `null`;

exports[`<ProfileEditView /> snapshots should render the details of the user 1`] = `null`;
